src/adapter/src/coord.rs

```diff
@@ -233,6 +233,34 @@ pub enum Message<T = mz_repr::Timestamp> {
     },
 }
 
+impl Message {
+    /// Returns a string to identify the kind of [`Message`], useful for logging.
+    pub const fn kind(&self) -> &'static str {
```

Comment:
Thoughts about using enum_kind and derive Debug instead? That'd eliminate typing here which seems convenient.

Reply:
We can do that, the reason I didn't because then we'd need to heap allocate a small string every time when using the `Debug` impl for the generated EnumKind, as opposed to getting a `&'static str` with this approach. Definitely a 🔥 _micro optimization_ 🔥 but I figured why not for something on the hot path like this. What do you think?

Reply:
Good point! Agree it's better as a static str because this is definitely a hot path.